File: NAFClient/src/shared/types/api/resourceRequestAllowance.ts
export type ResourceRequestAllowance = {
  id: number;
  resourceId: number;
  resourceName: string;
  locationId: number;
  locationName: string;
  allowanceDays: number;
};

export type CreateResourceRequestAllowanceDTO = {
  resourceId: number;
  locationId: number;
  allowanceDays: number;
};

export type UpdateResourceRequestAllowanceDTO = {
  allowanceDays: number;
};

File: NAFServer/src/Domain/Entities/ResourceRequestAllowance.cs
namespace NAFServer.src.Domain.Entities
{
    public class ResourceRequestAllowance
    {
        public int Id { get; set; }
        public int ResourceId { get; set; }
        public int LocationId { get; set; }
        public int AllowanceDays { get; set; }

        public Resource Resource { get; set; }
        public Location Location { get; set; }

        private ResourceRequestAllowance() { }

        public ResourceRequestAllowance(int resourceId, int locationId, int allowanceDays)
        {
            ResourceId = resourceId;
            LocationId = locationId;
            AllowanceDays = allowanceDays;
        }
    }
}
